pkg/controller/backup_configuration.go

```diff
@@ -0,0 +1,356 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/appscode/go/log"
+	workload_api "github.com/appscode/kubernetes-webhook-util/apis/workload/v1"
+	batch_util "github.com/appscode/kutil/batch/v1beta1"
+	core_util "github.com/appscode/kutil/core/v1"
+	"github.com/appscode/kutil/tools/queue"
+	api_v1beta1 "github.com/appscode/stash/apis/stash/v1beta1"
+	"github.com/appscode/stash/client/clientset/versioned/scheme"
+	"github.com/appscode/stash/pkg/docker"
+	"github.com/appscode/stash/pkg/eventer"
+	"github.com/appscode/stash/pkg/util"
+	"github.com/golang/glog"
+	"k8s.io/api/batch/v1beta1"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/reference"
+)
+
+func (c *StashController) initBackupConfigurationWatcher() {
+	c.bupcInformer = c.stashInformerFactory.Stash().V1beta1().BackupConfigurations().Informer()
+	c.bupcQueue = queue.New("BackupConfiguration", c.MaxNumRequeues, c.NumThreads, c.runBackupConfigurationInjector)
+	c.bupcInformer.AddEventHandler(&cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if b, ok := obj.(*api_v1beta1.BackupConfiguration); ok {
```

Comment:
Why are we writing `event` here? We can add validation check and event later.

Reply:
Removed event